lib/poxa/channel.ex

```diff
@@ -40,6 +40,23 @@ defmodule Poxa.Channel do
   end
 
   @doc """
+  Returns true if prefix matches channel
+
+  iex> Poxa.Channel.matches?("foo-bar", "foo-")
+  true
+  iex> Poxa.Channel.matches?("foo-bar", "bar-")
+  false
+  iex> Poxa.Channel.matches?("foo-bar", "")
+  false
+  """
+  def matches?(_, ""), do: false
+  def matches?(channel, prefix) do
+    base = byte_size(prefix)
+    <<match :: binary-size(base), _ :: binary>> = channel
```

Comment:
Here you can simply use `^prefix` instead of `match` and remove the match == prefix. The `^` operator will keep the binding of the original variable. This operator is called "pin operator": http://elixir-lang.org/getting-started/pattern-matching.html

So you could write this:

```elixir
  def matches?(_, ""), do: false
  def matches?(channel, prefix) do
    base = byte_size(prefix)
    case channel do
      <<^prefix :: binary-size(base), _ :: binary>> -> true
      _ -> false
    end
  end
```

The case is preferred in this case as if the match fails for some reason, it will always give `false`. One example that will explode this match: `Poxa.Channel.matches?("foo-bar", "foo-barrr")`